coreax/util.py

```diff
@@ -244,6 +245,41 @@ def solve_qp(kernel_mm: ArrayLike, gramian_row_mean: ArrayLike, **osqp_kwargs) -
     return sol.primal
 
 
+def sample_batch_indices(
+    random_key: KeyArrayLike,
+    data_size: int,
+    batch_size: int,
+    num_batches: int,
+) -> ArrayLike:
+    """
+    Sample an array of indices of size batch_size x num_batches.
+
+    Each column (batch) of the sampled array will contain unique elements. The largest
+    possible index sampled is dictated by data_size.
+
+    :param random_key: Key for random number generation
+    :param data_size: Size of the data we wish to sample from
+    :param batch_size: Size of the batch we wish to sample
+    :param num_batches: Number of batches to sample
+    :return: Array of batch indices of size `batch_size` x `num_batches`
+    """
+    if data_size < batch_size:
+        raise ValueError("data_size must be greater than or equal to batch_size")
+    if (data_size <= 0.0) or not isinstance(data_size, int):
+        raise ValueError("data_size must be a positive integer")
+    if (batch_size <= 0.0) or not isinstance(batch_size, int):
+        raise ValueError("batch_size must be a positive integer")
+    if (num_batches <= 0.0) or not isinstance(num_batches, int):
+        raise ValueError("num_batches must be a positive integer")
+
+    return permutation(
```

Comment:
I think this would be more idiomatic, and the logic easier to follow, if this were written:
```python
batch_keys = jr.split(key, num_batches)
batch_permutation = jax.vmap(jr.permutation, in_axes=(0, None))
return batch_permutation(batch_keys, data_size)
```
We don't need to set `independant=True`, because we only consider one vector at a time (inside the vmap).

Reply:
Much nicer! Updated the code above to ensure the array comes out the right size.